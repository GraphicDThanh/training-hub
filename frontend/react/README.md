# React Redux Training
[Plan doc](https://docs.google.com/document/d/1lYywN0RIhrwsyFZ8uPZuaN5QM78mYuXFD608vS8YEbM/edit?pli=1#)
## ES6
## React Basic
### Practice 1 - examples React
## React Advanced
### Practice 2
## Redux Basic
## Redux Advanced
## Redux Recipes
