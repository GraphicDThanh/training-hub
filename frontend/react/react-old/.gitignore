node_modules
out
dist
.DS_Store
.git
.docpad.db
vendor/
.vagrant
bower-components/
bower_components/
npm-debug.log
.gitignore~
vagrant/Vagrantfile~
build/
