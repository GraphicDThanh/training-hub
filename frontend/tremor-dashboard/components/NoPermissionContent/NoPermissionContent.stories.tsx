// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import NoPermissionContent from "./NoPermissionContent";

const meta = {
  title: "Components/Common/NoPermissionContent",
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "NoPermissionContent to apply for user not permission",
      },
    },
  },
  argTypes: {
    options: { description: "Option of select field" },
    className: { description: "Class name of select field" },
    label: { description: "Label of select field" },
  },
  component: NoPermissionContent,
} as Meta<typeof NoPermissionContent>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
