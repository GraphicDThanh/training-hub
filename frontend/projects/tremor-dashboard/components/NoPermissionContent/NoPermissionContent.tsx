import Link from "next/link";

import { Flex } from "@tremor/react";
import { MdOutlineSentimentDissatisfied } from "react-icons/md";

import { ROUTES } from "@/constants";

const NoPermissionContent = () => (
  <Flex flexDirection="col" className="inset-0 justify-center mt-48">
    <MdOutlineSentimentDissatisfied size="64" className="w-10 text-gray-400" />
    <h2 className="text-sm lg:text-lg text-center text-tremor-content-title dark:text-dark-tremor-content-title my-2">
      Apologies, but you require additional permissions to access this page.
    </h2>
    <Link
      href={ROUTES.HOME}
      className="text-white text-sm rounded-lg bg-gradient-pickled py-3 px-6 mt-8 border-none text-center">
      Back to Home
    </Link>
  </Flex>
);
export default NoPermissionContent;
