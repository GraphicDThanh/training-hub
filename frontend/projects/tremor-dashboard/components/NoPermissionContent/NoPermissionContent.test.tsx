import { render } from "@testing-library/react";

// Components
import NoPermissionContent from "./NoPermissionContent";

describe("NoPermissionContent Testing", () => {
  it("Should match snapshot", () => {
    const component = render(<NoPermissionContent />);
    expect(component).toMatchSnapshot();
  });
});
