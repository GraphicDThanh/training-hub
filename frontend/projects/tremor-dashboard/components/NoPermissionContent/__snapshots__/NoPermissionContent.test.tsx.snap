// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPermissionContent Testing Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tremor-Flex-root flex w-full flex-col items-center inset-0 justify-center mt-48"
      >
        <svg
          class="w-10 text-gray-400"
          fill="currentColor"
          height="64"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="64"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <circle
            cx="15.5"
            cy="9.5"
            r="1.5"
          />
          <circle
            cx="8.5"
            cy="9.5"
            r="1.5"
          />
          <path
            d="M12 14c-2.33 0-4.32 1.45-5.12 3.5h1.67c.69-1.19 1.97-2 3.45-2s2.75.81 3.45 2h1.67c-.8-2.05-2.79-3.5-5.12-3.5zm-.01-12C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <h2
          class="text-sm lg:text-lg text-center text-tremor-content-title dark:text-dark-tremor-content-title my-2"
        >
          Apologies, but you require additional permissions to access this page.
        </h2>
        <a
          class="text-white text-sm rounded-lg bg-gradient-pickled py-3 px-6 mt-8 border-none text-center"
          href="/analytics"
        >
          Back to Home
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tremor-Flex-root flex w-full flex-col items-center inset-0 justify-center mt-48"
    >
      <svg
        class="w-10 text-gray-400"
        fill="currentColor"
        height="64"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <circle
          cx="15.5"
          cy="9.5"
          r="1.5"
        />
        <circle
          cx="8.5"
          cy="9.5"
          r="1.5"
        />
        <path
          d="M12 14c-2.33 0-4.32 1.45-5.12 3.5h1.67c.69-1.19 1.97-2 3.45-2s2.75.81 3.45 2h1.67c-.8-2.05-2.79-3.5-5.12-3.5zm-.01-12C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <h2
        class="text-sm lg:text-lg text-center text-tremor-content-title dark:text-dark-tremor-content-title my-2"
      >
        Apologies, but you require additional permissions to access this page.
      </h2>
      <a
        class="text-white text-sm rounded-lg bg-gradient-pickled py-3 px-6 mt-8 border-none text-center"
        href="/analytics"
      >
        Back to Home
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
